src/mapping/cellranger_multi/script.py

```diff
@@ -116,7 +113,52 @@ def setup_logger():
                    if cellranger_param == "reference")
 LIBRARY_PARAMS = tuple(LIBRARY_CONFIG_KEYS.keys())
 SAMPLE_PARAMS = tuple(SAMPLE_PARAMS_CONFIG_KEYS.keys())
+HELPER_INPUT = {
+    'gex_input': 'Gene Expression',
+    'abc_input': 'Antibody Capture',
+    'cgc_input': 'CRISPR Guide Capture',
+    'mux_input': 'Multiplexing Capture',
+    'vdj_input': 'VDJ',
+    'vdj_t_input': 'VDJ-T',
+    'vdj_t_gd_input': 'VDJ-T-GD',
+    'vdj_b_input': 'VDJ-B',
+    'agc_input': 'Antigen Capture'
+}
+
 
+def infer_library_id_from_path(input_path: str) -> str:
+    match = re.match(fastq_regex, input_path)
+    assert match is not None, \
+        f"File name of '{input_path}' should match regex {fastq_regex}."
+    return match.group(1)
+
+def transform_helper_inputs(par: dict[str, any]) -> dict[str, any]:
+    helper_input = {
+        "input": [],
+        "library_id": [],
+        "library_type": []
+    }
+    for input_type, library_type in HELPER_INPUT.items():
+        if par[input_type]:
+            par[input_type] = resolve_input_directories_to_fastq_paths(par[input_type])
+
+            library_ids = [
```

Comment:
Do we need to check for uniqueness of library IDs here?

Reply:
Added an assert: https://github.com/openpipelines-bio/openpipeline/blob/cellranger_multi_sample_sheets/src/mapping/cellranger_multi/script.py#L158

Reply:
Thanks 👍